### PR TITLE
Summer Open

### PR DESCRIPTION
**Pull request by summerwinter7**
I altered the Conversation Servlet so the page will only show a list of the public conversations, not the private ones. This will be the page that is shown when no one is logged in, or if the user logged in is not a part of any private conversations, and is a stepping stone to the final version, which will show all of the public conversations as well as the private ones that the user is a member of. I also added a test to make sure the added variable publicConversations only keeps track of the public ones.

### CODE REVIEW

File: src/main/java/codeu/controller/ConversationServlet.java [hunk 1]
```diff
@@ -70,8 +70,8 @@ void setConversationStore(ConversationStore conversationStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    List<Conversation> conversations = conversationStore.getAllConversations();
-    request.setAttribute("conversations", conversations);
+	List<Conversation> publicConversations = conversationStore.getAllPublicConversations();
+	request.setAttribute("publicConversations", publicConversations);
     request.getRequestDispatcher("/WEB-INF/view/conversations.jsp").forward(request, response);
```
Comment (simin4950) [inline src/main/java/codeu/controller/ConversationServlet.java:75]: For the group conversations, we could create a list that is groupConversations using this same idea.

File: src/main/java/codeu/model/store/basic/ConversationStore.java [hunk 2]
```diff
@@ -125,6 +136,12 @@ public Conversation getConversationWithID(UUID id) {
 	/** Sets the List of Conversations stored by this ConversationStore. */
 	public void setConversations(List<Conversation> conversations) {
 		this.conversations = conversations;
+		this.publicConversations = new ArrayList<Conversation>();
+		for (Conversation conversation : this.conversations) {
+			if (conversation.getIsPublic()) {
+				publicConversations.add(conversation);
+			}
+		}
 	}
```
Comment (simin4950) [inline src/main/java/codeu/model/store/basic/ConversationStore.java:146]: This file looks great; however, I was a little confused on why the "conversations" list is still kept. Is there a reason? We could use this list to include all conversations (group or public).

File: src/main/webapp/WEB-INF/view/conversations.jsp [hunk 3]
```diff
@@ -60,10 +60,11 @@
 
     <h1>Conversations</h1>
 
+	<h4>Public Conversations (Anyone can view):</h4>
     <%
-    List<Conversation> conversations =
```
Comment (simin4950) [inline src/main/webapp/WEB-INF/view/conversations.jsp:64]: I see here you have removed the "conversations" list but have kept it in the ConversationStore.

**Review by simin4950 — COMMENTED** [review]
So the files look good; however, in the ConversationStore file you have kept the "conversation" list which may be fine--just a little confused since you removed it in the other files. Is there another purpose for it?

File: src/main/java/codeu/controller/ConversationServlet.java (hunk 1 above)
Comment (summerwinter7) [reply src/main/java/codeu/controller/ConversationServlet.java:75]: Yeah I was thinking the same thing! That's what I was going to do in my next PR.

File: src/main/java/codeu/model/store/basic/ConversationStore.java (hunk 2 above)
Comment (summerwinter7) [reply src/main/java/codeu/model/store/basic/ConversationStore.java:146]: I may not need it in the end, but I've kept it for now to keep track of all of the conversations, so group and public. Right now the conversation page only shows the public conversations, which is why I only use the publicConversations variable, but next I will also add the group/private conversations, so I think I'll need the conversations list there.

**Review by simin4950 — APPROVED** [review]
I could make the groupConversations variables in my PR if that´s okay with you! Otherwise, everything looks good!

**Pull request merged**